File: AkkaUnitTest/src/main/java/org/akka/essentials/unittest/actors/FilteringActor.java
package org.akka.essentials.unittest.actors;

import akka.actor.ActorRef;
import akka.actor.UntypedActor;

public class FilteringActor extends UntypedActor {
	ActorRef next;

	public FilteringActor(ActorRef next) {
		this.next = next;
	}

	@Override
	public void onReceive(Object message) throws Exception {
		if (message instanceof String)
			next.tell(message);
	}
}

File: AkkaUnitTest/src/main/scala/org/akka/essentials/unittest/example/FilteringActor.scala
package org.akka.essentials.unittest.example
import akka.actor.actorRef2Scala
import akka.actor.Actor
import akka.actor.ActorRef

class FilteringActor(next: ActorRef) extends Actor {
  def receive: Receive = {
    case message: String => next ! message
  }
}